### PR TITLE
[Enhancement](meta) add err type/code/msg in audit event

### PR DESCRIPTION
**Pull request by chovy-3012**
# Proposed changes

Print "err msg" "err type"  and "err code" when sql execute failed .

Add "err msg" "err code" and "err type" in audit event .

Help developers diagnose the causes of failures.

For example:
```shell
errCode = 2, detailMessage = failed to initialize storage reader. tablet=76927688.1343432537.bd4a545605c7d580-8c277220384848b0, res=-230, backend=xxxxx
```

## Problem summary

Get more error information to help diagnose when sql execute failed .

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -164,6 +164,13 @@ private void auditAfterExec(String origStmt, StatementBase parsedStmt, Data.PQue
                 .setQueryId(ctx.queryId() == null ? "NaN" : DebugUtil.printId(ctx.queryId()))
                 .setTraceId(spanContext.isValid() ? spanContext.getTraceId() : "");
 
+        // audit error msg
+        if (ctx.getState().getStateType() == MysqlStateType.ERR) {
+            ctx.getAuditEventBuilder().setErrType(ctx.getState().getErrType().name())
+                .setErrCode(ctx.getState().getErrorCode().name())
```
Comment (cambyzju) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:170]: In the following two cases, ErrType, ErrCode and ErrMsg will be empty or other values? 
1. if we got MysqlStateType.ERR, but getState().getErrorCode() not set;
2. the error code returned by getState().getErrorCode() not registered;

Comment (chovy-3012) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:170]: Yes, this case may happed and cause null point exception in `ctx.getState().getErrorXXX().name()`.
Do you suggest adding a empty/null value judgment ?

Comment (cambyzju) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:170]: Yes I think it is necessary.

Comment (chovy-3012) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:170]: Okeydokey

**Review by morningman — APPROVED** [review]
LGTM

**Pull request closed without merging**